src/rdkafka_broker.c

```diff
@@ -3462,9 +3462,21 @@ rd_kafka_fetch_reply_handle (rd_kafka_broker_t *rkb,
 		rd_kafka_op_throttle_time(rkb, rkb->rkb_rk->rk_rep,
 					  Throttle_Time);
 	}
-
+#if WITH_ZSTD
+        if (rd_kafka_buf_ApiVersion(request) >= 10) {
+            int16_t error_code;
+            int32_t session_id;
+            rd_kafka_buf_read_i16(rkbuf, &error_code);
+            if (unlikely(error_code)) {
+                // Caller is printing and handling the error
+                return error_code;
+            }
+            rd_kafka_buf_read_i32(rkbuf, &session_id);
+        }
+#endif
 	rd_kafka_buf_read_i32(rkbuf, &TopicArrayCnt);
 	/* Verify that TopicArrayCnt seems to be in line with remaining size */
+        /* TODO: updated below check for v10 response?*/
```

Comment:
As long as v10 isn't smaller than v4, then this should be fine. Remove TODO comment.